frontend/src/components/navigation/FileSelector.tsx

```diff
@@ -0,0 +1,288 @@
+		}
+	};
+
+	// Function to fetch children of a folder
+	async function fetchFolderFiles(
+		datasetid: string,
+		folderId: string | undefined
+	) {
+		try {
+			const response =
+				await V2.DatasetsService.getDatasetFoldersAndFilesApiV2DatasetsDatasetIdFoldersAndFilesGet(
+					datasetid,
+					folderId,
+					// TODO: Remove hardcoded values
+					0,
+					3000
+				);
+			const data = response.data;
+			const FSItems: FSItem[] =
+				data !== undefined
+					? data.map((FSItem: any) => ({
+						datasetId: datasetid,
+						id: FSItem.id,
+						label: FSItem.name,
+						children: FSItem.object_type === "folder" ? [] : undefined,
+						type: FSItem.object_type,
+						content_type:
+							FSItem.object_type !== "folder"
+								? FSItem.content_type.main_type
+								: undefined
+					}))
+					: [];
+			return FSItems;
+		} catch (error) {
+			console.error("Error fetching folders and files", error);
+		}
+	}
+
+	// Function to handle selection of folder or file
+	const onSelect = () => {
+		if (isFolder) {
+			if (!expanded) {
+				fetchFolderFiles(item.datasetId, item.id).then((data) => {
+					setChildren(data);
+				});
+			}
+			setExpanded(!expanded);
+		} else {
+			if (item.id !== undefined) {
+				onSelectFile(item.id, item.label);
+			}
+		}
+	};
+
+	return (
+		<List disablePadding>
+			{/* Indentation of item proportional to depth */}
+			<ListItem
+				sx={{
+					pl: depth * 2,
+					borderBottom: "none",
+					py: 0.5,
+					"&:hover": {
+						backgroundColor: "rgba(0, 0, 0, 0.1)", // or any other color
+						cursor: "pointer"
+					}
+				}}
+				onClick={onSelect}
+			>
+				<ListItemIcon sx={{ minWidth: "auto", mr: 1 }}>
+					<IconButton
+						size="small"
+						sx={{ visibility: isFolder ? "visible" : "hidden" }}
+					>
+						<ExpandMoreIcon
+							style={{
+								transform: expanded ? "rotate(0deg)" : "rotate(-90deg)"
+							}}
+						/>
+					</IconButton>
+				</ListItemIcon>
+				<ListItemIcon sx={{ minWidth: "auto" }}>{getIcon()}</ListItemIcon>
+				<ListItemText primary={item.label} />
+			</ListItem>
+			<Collapse in={expanded} timeout="auto" unmountOnExit>
+				{isFolder && (
+					<Box sx={{ ml: 2 }}>
+						{children?.map((child) => (
+							<RecursiveComponent
+								key={child.id}
+								item={child}
+								depth={depth + 1}
+								onSelectFile={onSelectFile}
+							/>
+						))}
+					</Box>
+				)}
+			</Collapse>
+		</List>
+	);
+};
+
+const FileSystemViewer: React.FC<{
+	onSelectFile: (fileId: string, fileName: string) => void;
+}> = ({ onSelectFile }) => {
+	const dispatch = useDispatch();
+	const datasets = useSelector((state: any) => state.dataset.datasets);
+	const [FSItems, setFSItems] = useState<FSItem[]>([]);
+
+	// API function call to get Datasets
+	const listDatasets = (skip?: number, limit?: number, mine?: boolean) => {
+		dispatch(fetchDatasets(skip, limit, mine));
+	};
+
+	// Fetch datasets on component mount
+	useEffect(() => {
+		listDatasets(0, 10, true);
```

Comment:
You are fixing the dataset to be 10? Could you add a todo here to address dataset pagination.